drivers/serial/uart_nrfx_uarte.c

```diff
@@ -850,9 +857,14 @@ static int uarte_nrfx_rx_enable(const struct device *dev, uint8_t *buf,
 	 * way but timeouts would always occur later than expected,  most likely
 	 * after ~3 ticks.
 	 */
+#ifdef UARTE_HW_ASYNC
+	data->async->rx_timeout_slab = timeout / RX_TIMEOUT_DIV;
```

Comment:
Only this one line should stay here. Everything else in the surrounding `#ifdef` and the preceding comment should be removed.